### PR TITLE
Add release.yaml and RELEASE.md

### PR DESCRIPTION
**Pull request by criccomini**
This PR adds a release.yaml Github action to make it easier to publish SlateDB releases. The action is triggered through the Github actions page. To run it, SlateDB committers input a version number and click "run workflow". The workflow will:

1. Verify that the version adheres to the semantic versioning format.
2. Check out the latest code from the `main` branch.
3. Update the version in Cargo.toml to the specified version.
4. Commit the changes and push to the `main` branch.
5. Create a Github release with the specified version and auto-generated release notes.
6. Publish a release to crates.io.

I've also added a RELEASE.md to describe this process.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Releasing SlateDB
+
+SlateDB releases are published using a Github release action defined in `.github/workflows/release.yml`. To create a new release:
+
+1. Go to the [release action page](https://github.com/slatedb/slatedb/actions/workflows/pr.yaml)
```
Comment (calavera) [inline RELEASE.md:5]: is the link correct? shouldn't it be `release.yaml`?

Comment (criccomini) [reply RELEASE.md]: Ah good catch. Fixed: e3c39cc

**Review by calavera — APPROVED** [review]
This is cool, I might steal it for my own projects 😆

**Comment by criccomini** [comment]
Merging optiomistically

**Pull request merged**